brainlit/algorithms/generate_fragments/spline_fxns.py

```diff
@@ -0,0 +1,296 @@
+import operator
+from scipy.interpolate import BSpline, splev
+import numpy as np
+from typing import Iterable, Union, Tuple
+from brainlit.utils.util import (
+    check_type,
+    check_size,
+    check_precomputed,
+    check_iterable_type,
+    check_iterable_nonnegative,
+)
+
+
+def speed(
+    x: np.ndarray,
+    t: np.ndarray,
+    c: np.ndarray,
+    k: np.integer,
```

Comment:
What do you think about combining these inputs into a tuple tck, so we can match the input format of[ scipy.splev](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.splev.html)?

Reply:
My main reasons not to do it are:

1. From [scipy documentation](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.splev.html):
> Manipulating the tck-tuples directly is not recommended. In new code, prefer using BSpline objects.

2. Using `scipy.interpolate.BSpline` does not require us to use `scipy.interpolate.splev` anymore because we can call `spline(x)` to evaluate it.

3. Splitting the `tck` couple is consistent with `scipy.interpolate.BSpline` methods.